orchagent/saihelper.cpp

```diff
@@ -803,3 +831,270 @@ void handleSaiFailure(bool abort_on_failure)
         abort();
     }
 }
+
+
+static inline void initSaiRedisCounterEmptyParameter(sai_s8_list_t &sai_s8_list)
+{
+    sai_s8_list.list = nullptr;
+    sai_s8_list.count = 0;
+}
+
+static inline void initSaiRedisCounterEmptyParameter(sai_redis_flex_counter_group_parameter_t &flex_counter_group_param)
+{
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.poll_interval);
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.operation);
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.stats_mode);
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.plugin_name);
+    initSaiRedisCounterEmptyParameter(flex_counter_group_param.plugins);
+}
+
+static inline void initSaiRedisCounterParameterFromString(sai_s8_list_t &sai_s8_list, const std::string &str)
+{
+    if (str.length() > 0)
+    {
+        sai_s8_list.list = (int8_t*)const_cast<char *>(str.c_str());
+        sai_s8_list.count = (uint32_t)str.length();
+    }
+    else
+    {
+        sai_s8_list.list = nullptr;
+        sai_s8_list.count = 0;
+    }
+}
+
+static inline void notifySyncdCounterOperation(bool is_gearbox, const sai_attribute_t &attr)
+{
+    if (sai_switch_api == nullptr)
+    {
+        // This can happen during destruction of the orchagent daemon.
+        SWSS_LOG_ERROR("sai_switch_api is NULL");
+        return;
+    }
+
+    if (!is_gearbox)
+    {
+        sai_switch_api->set_switch_attribute(gSwitchId, &attr);
+    }
+    else
+    {
+        for (auto gearbox_oid : gGearboxOids)
+        {
+            sai_switch_api->set_switch_attribute(gearbox_oid, &attr);
+        }
+    }
+}
+
+static inline void operateFlexCounterDbSingleField(std::vector<FieldValueTuple> &fvTuples,
+                                            const string &field, const string &value)
+{
+    if (!field.empty() && !value.empty())
```

Comment:
is this prevention here for a reason? ot at some point OA can actually populate empty counter? if any of those is empty this should be logged as ERROR

Reply:
This is also for mock test.
The common logic to install plugin is like this:
1. load lua plugin and generate SHA code
2. install the lua plugin into {`plugin name`: `plugin SHA code`} the flex counter group

In mock test scenario, it fails to load the Lua plugin, which leaves the SHA code empty string. But we still want the lua plugin to be installed into the flex counter group. By doing so, we can verify whether the plugin name is correct and meet the coverage requirement.

In production scenario, we will check whether either is empty, if so, we will not install the plugin. This is just a production and can not happen.